### PR TITLE
Use NamedArea in each dataset

### LINKED ISSUE #106: rnpn issues
**1. Bug in Example in docstring:**
```python
Example:
        # Load dataset
        dataset = RNPN(
            species_ids=[36],  # Syringa vulgaris / common lilac
            phenophase_ids=[483],  # Leaves
            years=[2010, 2012],
        )
```

Both species_ids and phenophase_ids should be defined as dictionary e.g.: `species_ids={"name": "Syringa", "items":[36]}, phenophase_ids={"name": "leaves", "items":[483]}`

**2. bug in load:** If data is not available for one year, like the example below, `load` function reruns error:
```python
dataset = RNPN(species_ids={"name": "Syringa", "items":[36]}, phenophase_ids={"name": "leaves", "items":[483]}, years=[2012, 2013])
dataset.download()

downloading /tmp/data/rnpn/rnpn_npn_data_y_2012_Syringa_leaves.csv
 Found 115 records...NULL
downloading /tmp/data/rnpn/rnpn_npn_data_y_2013_Syringa_leaves.csv
 Found 0 records...NULL
```

```python
dataset.load()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/sarah/GitHub/springtime/src/springtime/datasets/rnpn.py", line 96, in load
    [
  File "/home/sarah/GitHub/springtime/src/springtime/datasets/rnpn.py", line 97, in <listcomp>
    pd.read_csv(self._filename(year))
  File "/home/sarah/.local/share/hatch/env/virtual/springtime/x0W6Y1ik/springtime/lib/python3.9/site-packages/pandas/util/_decorators.py", line 211, in wrapper
    return func(*args, **kwargs)
  File "/home/sarah/.local/share/hatch/env/virtual/springtime/x0W6Y1ik/springtime/lib/python3.9/site-packages/pandas/util/_decorators.py", line 331, in wrapper
    return func(*args, **kwargs)
  File "/home/sarah/.local/share/hatch/env/virtual/springtime/x0W6Y1ik/springtime/lib/python3.9/site-packages/pandas/io/parsers/readers.py", line 950, in read_csv
    return _read(filepath_or_buffer, kwds)
  File "/home/sarah/.local/share/hatch/env/virtual/springtime/x0W6Y1ik/springtime/lib/python3.9/site-packages/pandas/io/parsers/readers.py", line 605, in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
  File "/home/sarah/.local/share/hatch/env/virtual/springtime/x0W6Y1ik/springtime/lib/python3.9/site-packages/pandas/io/parsers/readers.py", line 1442, in __init__
    self._engine = self._make_engine(f, self.engine)
  File "/home/sarah/.local/share/hatch/env/virtual/springtime/x0W6Y1ik/springtime/lib/python3.9/site-packages/pandas/io/parsers/readers.py", line 1735, in _make_engine
    self.handles = get_handle(
  File "/home/sarah/.local/share/hatch/env/virtual/springtime/x0W6Y1ik/springtime/lib/python3.9/site-packages/pandas/io/common.py", line 856, in get_handle
    handle = open(
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/data/rnpn/rnpn_npn_data_y_2013_Syringa_leaves.csv
```

### PR DESCRIPTION
**Pull request by sverhoeven**
Refs #92 
Fixes #106

### CODE REVIEW

File: src/springtime/datasets/rnpn.py [hunk 1]
```diff
@@ -96,6 +97,7 @@ def load(self):
             [
                 pd.read_csv(self._filename(year))
                 for year in range(self.years[0], self.years[1] + 1)
+                if self._filename(year).exists()
```
Comment (Peter9192) [inline src/springtime/datasets/rnpn.py:100]: Skipping without notice?

File: src/springtime/datasets/e_obs.py [hunk 2]
```diff
@@ -178,17 +179,32 @@ class EOBSBoundingBox(EOBS):
     Fetches complete grid from
     https://surfobs.climate.copernicus.eu/dataaccess/access_eobs.php .
 
+    Example:
+
+        To load coarse mean temperature around amsterdam from 2002 till 2002::
```
Comment (Peter9192) [inline src/springtime/datasets/e_obs.py:184]:
```suggestion
        To load coarse mean temperature around amsterdam from 2002 till 2002:
```

File: src/springtime/utils.py [hunk 3]
```diff
@@ -7,18 +7,25 @@
...
 
-from typing import Sequence, Tuple
+from typing import NamedTuple, Sequence
 from pydantic import BaseModel, validator
 from shapely.geometry import Polygon
 
 logger = getLogger(__name__)
 
 
+class BoundingBox(NamedTuple):
```
Comment (Peter9192) [inline src/springtime/utils.py:17]: 👍

**Review by Peter9192 — APPROVED** [review]
Thanks, this is good!

File: src/springtime/datasets/rnpn.py (hunk 1 above)
Comment (sverhoeven) [reply src/springtime/datasets/rnpn.py:100]: If a year fails then  will raise exception at https://github.com/phenology/springtime/pull/107/files#diff-4b359fbfc7933d8272d65732a55d7a277f4307c363ebeb1c382616e8f33296bcR130

File: src/springtime/datasets/e_obs.py (hunk 2 above)
Comment (sverhoeven) [reply src/springtime/datasets/e_obs.py:184]: part of google docstring style to render below as code block

**Pull request merged**